packages/frontend/src/components/FlowRow/FlowContextMenu.tsx

```diff
@@ -0,0 +1,162 @@
+import { IFlow } from '@plumber/types'
+
+import { MouseEvent, useCallback, useRef } from 'react'
+import { BiDotsHorizontalRounded } from 'react-icons/bi'
+import { BsTrash } from 'react-icons/bs'
+import { MdOutlineRemoveRedEye } from 'react-icons/md'
+import { Link } from 'react-router-dom'
+import { useMutation } from '@apollo/client'
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogContent,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogOverlay,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  useDisclosure,
+} from '@chakra-ui/react'
+import {
+  Button,
+  IconButton,
+  TouchableTooltip,
+  useToast,
+} from '@opengovsg/design-system-react'
+import * as URLS from 'config/urls'
+import { DELETE_FLOW } from 'graphql/mutations/delete-flow'
+
+interface FlowContextMenuProps {
+  flow: IFlow
+}
+
+export default function FlowContextMenu(props: FlowContextMenuProps) {
+  const { flow } = props
+
+  // dialog control
+  const {
+    isOpen: isDialogOpen,
+    onOpen: onDialogOpen,
+    onClose: onDialogClose,
+  } = useDisclosure()
+
+  // menu control
+  const {
+    isOpen: isMenuOpen,
+    onToggle: onMenuToggle,
+    onClose: onMenuClose,
+  } = useDisclosure()
+
+  const cancelRef = useRef<HTMLButtonElement>(null)
+  const toast = useToast()
+  const [deleteFlow, { loading: isDeletingFlow }] = useMutation(DELETE_FLOW)
+  const flowTransfer = flow?.pendingTransfer
```

Comment:
nit: maybe something like `hasPendingTransfer`?

Reply:
oh but i need this as a `FlowTransfer` to access like the oldOwner or newOwner so I thought it was semantically right to just use `pendingTransfer`, feel free to ask me irl if u disagree, thanks!